Plugin/PaymentInformationManagement.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2019 Adyen BV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Plugin;
+
+use Adyen\Payment\Observer\AdyenHppDataAssignObserver;
+use Adyen\Payment\Observer\AdyenCcDataAssignObserver;
+use Magento\Vault\Model\Ui\VaultConfigProvider;
+
+class PaymentInformationManagement
```

Comment:
The main idea is to always use the original/existing endpoints instead of exposing new ones and the set-payment-information endpoint looks just perfect for this usage (which saves the payment information but doesn't place the order) and in the flow we need to add the new 3ds2 functionality.

The other approach could be exposing a new endpoint and using that but the Magento community suggested to use the existing default ones